src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,665 @@
+            Discard data prior to this specified time as 'equilibration' data. Units
+            picoseconds. Default: 0.
+        uncorr : str
+            The observable to be used for the autocorrelation analysis; 'dhdl'
+            (obtained as a sum over those energy components that are changing).
+            Specify as `None` will not uncorrelate the data. Default: `dhdl`.
+        threshold : int
+            Proceed with correlated samples if the number of uncorrelated samples is
+            found to be less than this number. If 0 is given, the time series
+            analysis will not be performed at all. Default: 50.
+        methods : str
+            A list of the methods to esitimate the free energy with. Default:
+            `('mbar', 'bar', 'ti')`.
+        overlap : str
+            The filename for the plot of overlap matrix. Default: `O_MBAR.pdf`.
+        breakdown : bool
+            Plot the free energy differences evaluated for each pair of adjacent
+            states for all methods, including the dH/dlambda curve for TI. Default:
+            True.
+        forwrev : int
+            Plot the free energy change as a function of time in both directions,
+            with the specified number of points in the time plot. The number of time
+            points (an integer) must be provided. Specify as `None` will not do
+            the convergence analysis. Default: 10.
+
+        Attributes
+        ----------
+        summary : Dataframe
+            The summary of the free energy estimate.
+        convergence : DataFrame
+            The summary of the convergence results.
+        '''
+        self.read()
+
+        if uncorr is not None:
+            self.preprocess(skiptime=skiptime, uncorr=uncorr,
+                               threshold=threshold)
+        if methods is not None:
+            self.estimate(methods)
+            self.generate_result()
+
+        if overlap is not None:
+            ax = self.plot_overlap_matrix(overlap)
+            plt.close(ax.figure)
+
+        if breakdown:
+            ax = self.plot_ti_dhdl()
+            plt.close(ax.figure)
+            fig = self.plot_dF_state()
+            plt.close(fig)
+            fig = self.plot_dF_state(dF_state='dF_state_long.pdf',
+                                    orientation='landscape')
+            plt.close(fig)
+
+        if forwrev is not None:
+            ax = self.check_convergence(forwrev, estimator='autombar',
+                                        dF_t='dF_t.pdf')
+            plt.close(ax.figure)
+
+
+    def update_units(self, units):
+        '''Update the unit.
+
+        Parameters
+        ----------
+        units : str
+            The unit used for printing and plotting results. {'kcal/mol',
+            'kJ/mol', 'kT'}
+
+        '''
+        if units is not None:
+            self.logger.info(f'Set unit to {units}.')
+            self.units = units
+        else: # pragma: no cover
+            pass
```

Comment:
Instead of this if statement, make `units=None` a kwarg.

Reply:
@orbeckst Sorry I wonder what do you mean?